### PR TITLE
Expose feature and label transformer in automl.py

### LINKED ISSUE #979: expose feature & label transformers in automl class object
Hi, thanks for the great library.

I'm trying to save an AutoML object's feature & label pre-processors/transformers, so that they can be run offline as part of a data pipeline.

Currently, I'm using code like this:

```
hyperparams, estimator_class, X_transformed, y_transformed, feature_transformer, label_transformer = preprocess_and_suggest_hyperparams(
    "classification", X_train, y_train, "lgbm"
)
```
but is there a way to get the feature_transformer and label_transformer exposed as an automl property, so I can just use automl?

### PR DESCRIPTION
**Pull request by JaneIllario**
## Why are these changes needed?

Exposes additional field in AutoML to eliminate a call to "preprocess_and_suggest_hyperparams" as the transformers are already accessible through the object

## Related issue number

Closes #979

## Checks

- [X] I've used [pre-commit](https://microsoft.github.io/FLAML/docs/Contribute#pre-commit) to lint the changes in this PR, or I've made sure [lint with flake8](https://github.com/microsoft/FLAML/blob/816a82a1155b4de4705b21a615ccdff67c6da379/.github/workflows/python-package.yml#L54-L59) output is two 0s.
- [ ] I've included any doc changes needed for https://microsoft.github.io/FLAML/. See https://microsoft.github.io/FLAML/docs/Contribute#documentation to build and test documentation locally.
- [X] I've added tests (if relevant) corresponding to the changes introduced in this PR.
- [X] I've made sure all auto checks have passed.

### CODE REVIEW

**Review by qingyun-wu — APPROVED** [review]
Looks good to me. Thanks!

**Comment by JaneIllario** [comment]
@skzhang1 @thinkall tagging bc Qingyun added you to the previous PR. Please take a look when you have a moment

**Review by thinkall — APPROVED** [review]
LGTM. Thanks.

File: flaml/automl/automl.py [hunk 1]
```diff
@@ -490,6 +490,16 @@ def save_best_config(self, filename):
         with open(filename, "w") as f:
             json.dump(best, f)
 
+    @property
+    def feature_transformer(self):
+        """Returns AutoML Transformer"""
+        return self._transformer
```
Comment (sonichi) [inline flaml/automl/automl.py:496]:
```suggestion
        return getattr(self, "_transformer", None)
```

File: flaml/automl/automl.py [hunk 2]
```diff
@@ -490,6 +490,16 @@ def save_best_config(self, filename):
...
+
+    @property
+    def label_transformer(self):
+        """Returns AutoML label transformer"""
+        return self._label_transformer
```
Comment (sonichi) [inline flaml/automl/automl.py:501]:
```suggestion
        return getattr(self, "_label_transformer", None)
```
In case the attribute is not set.

**Review by sonichi — COMMENTED** [review]
A minor suggestion to avoid undefined attribute error.

**Comment by JaneIllario** [comment]
@qingyun-wu Hi Qingyun! I'm having some trouble with the build checks, and am not sure why they're failing -- any ideas on what to try?

**Review by sonichi — APPROVED** [review]
I guess the build error is related to https://github.com/microsoft/FLAML/pull/995

**Pull request merged**